modules/navier_stokes/test/tests/finite_volume/wcns/materials/1d_test_cpT.i

```diff
@@ -0,0 +1,231 @@
+  []
+  [u_pressure]
+    type = INSFVMomentumPressure
+    variable = vel_x
+    momentum_component = 'x'
+    pressure = pressure
+  []
+
+  [temp_conduction]
+    type = FVDiffusion
+    coeff = 'k'
+    variable = T
+  []
+  [temp_advection]
+    type = INSFVEnergyAdvection
+    variable = T
+  []
+  [source]
+    type = FVBodyForce
+    variable = T
+    function = source_func
+  []
+[]
+
+[FVBCs]
+  [inlet-u]
+    type = INSFVInletVelocityBC
+    boundary = 'left'
+    variable = vel_x
+    functor = ${bulk_u}
+  []
+  [inlet_T]
+    type = FVDirichletBC
+    variable = T
+    boundary = 'left'
+    value = ${T_in}
+  []
+
+  [outlet_p]
+    type = INSFVOutletPressureBC
+    boundary = 'right'
+    variable = pressure
+    function = ${p_ref}
+  []
+[]
+
+[Functions]
+  [source_func]
+    type = ParsedFunction
+    expression = '${q_source}'
+  []
+[]
+
+[FunctorMaterials]
+  [converter_to_regular_T]
+    type = FunctorADConverter
+    ad_props_in = 'T'
+    reg_props_out = 'T_nAD'
+  []
+  [ins_fv]
+    type = INSFVEnthalpyFunctorMaterial
+    temperature = 'T'
+    rho = 'rho'
+    cp = 'cp'
+
+    assume_constant_cp = false
+    h_in = 'h'
+    # fp = 'fp'
+    # pressure = 'pressure'
+  []
+
+  [rho]
+    type = ADParsedFunctorMaterial
+    property_name = 'rho'
+    expression = '${rho}'
+  []
+  [mu]
+    type = ADParsedFunctorMaterial
+    property_name = 'mu'
+    expression = '4.5e-3'
+  []
+  [k]
+    type = ADParsedFunctorMaterial
+    property_name = 'k'
+    expression = '0.7'
+  []
+  [h]
+    type = ADParsedFunctorMaterial
+    property_name = 'h'
+    functor_names = 'T ${A_cp} ${B_cp}'
+    functor_symbols = 'T A_cp B_cp'
+    expression = 'A_cp * T + B_cp * T * T / 2'
+  []
+  [cp]
+    type = ADParsedFunctorMaterial
+    property_name = 'cp'
+    functor_names = 'T ${A_cp} ${B_cp}'
+    functor_symbols = 'T A_cp B_cp'
+    expression = 'A_cp+B_cp*T'
+  []
+[]
+
+[Executioner]
+  type = Steady
+  solve_type = 'NEWTON'
+  petsc_options_iname = '-pc_type -pc_factor_shift_type'
+  petsc_options_value = 'lu       NONZERO'
+
+  nl_abs_tol = 1e-9
+  nl_max_its = 50
+  line_search = 'none'
+
+  automatic_scaling = true
+  off_diagonals_in_auto_scaling = true
+[]
+
+[Postprocessors]
+  [H_in]
+    type = VolumetricFlowRate
+    vel_x = 'vel_x'
+    advected_quantity = 'rho_h'
+    boundary = 'left'
+  []
+  [H_out]
+    type = VolumetricFlowRate
+    vel_x = 'vel_x'
+    advected_quantity = 'rho_h'
+    boundary = 'right'
+  []
+  [Q]
+    type = FunctionElementIntegral
+    function = 'source_func'
+    execute_on = 'initial'
+  []
+  [balance_in_percent]
+    type = ParsedPostprocessor
+    expression = '(H_out + H_in - Q) / H_in * 100'
+    pp_names = 'H_in H_out Q'
+  []
+
+  [T_out]
+    type = SideAverageValue
+    variable = T
+    boundary = 'right'
+  []
+  [T_analytical]
+    type = Receiver
+    default = ${fparse (-A_cp+sqrt(A_cp^2-2*B_cp*(-q_source/rho/bulk_u*L-A_cp*T_in-B_cp/2*T_in*T_in)))/B_cp}
```

Comment:
I verified by hand calculation that this analytical solution is correct. However, please add a couple comments of the steps of the derivation for future reference.

Reply:
Ramiro did the derivation. @freiler do you want to add that as a suggestion?